website/util/time.py

```diff
@@ -10,7 +11,25 @@ def get_timestamp():
 def throttle_period_expired(timestamp, throttle):
     if not timestamp:
         return True
-    elif isinstance(timestamp, datetime.datetime):
-        return (datetime.datetime.utcnow() - timestamp).total_seconds() > throttle
+    elif isinstance(timestamp, datetime):
+        return (datetime.utcnow() - timestamp).total_seconds() > throttle
     else:
         return (get_timestamp() - timestamp) > throttle
+
+
+def generate_expiration_time(duration, base=datetime.utcnow(), utc=False):
+    """
+    Generate an expiration timestamp
+    :param duration: duration in seconds
+    :param base: optional base timestamp
+    :param utc: optional boolean flag which determines return type
+    :return: a timestamp object (default) or a utc string (if utc is set to True)
+    """
+    if not base:
+        base = datetime.utcnow()
+    elif not isinstance(base, datetime):
```

Comment:
This branch doesn't seem necessary. The user should only pass either a datetime or `None`.